src/features/cart/cartSlice.ts

```diff
@@ -0,0 +1,43 @@
+import {createSlice, PayloadAction} from "@reduxjs/toolkit";
+import {ProductType} from "@/features/product/productSlice";
+
+
+const initialState: ProductType[] = [];
+
+export const cartSlice = createSlice({
+        name: 'products',
+        initialState,
+        reducers: {
+            addToCart: (state, action: PayloadAction<{ product: ProductType, quantity: number }>) => {
+                const index = state.map(product => product.id).indexOf(action.payload.product.id);
+                if (state.length === 0) {
+                    const addedProduct = action.payload.product;
+                    const addedProductCopy = {...addedProduct};
+                    addedProductCopy.quantity = action.payload.quantity
+                    state.push(addedProductCopy);
+
+                } else {
+                    if (state.map(product => product.id).includes(action.payload.product.id)) {
+                        state[index].quantity += action.payload.quantity
+                    } else {
```

Comment:
too much branches in this method. it looks like it can be simplified to just
```ts
if(index > -1){
  state[index].quantity += action.payload.quantity
} else {
  state.push({ ...action.payload.product, quantity: action.payload.quantity })
}
```